core/src/main/scala/org/apache/spark/util/collection/WritablePartitionedPairCollection.scala

```diff
@@ -94,3 +83,20 @@ private[spark] trait WritablePartitionedIterator {
 
   def nextPartition(): Int
 }
+
+/**
+ * Default implementation of `WritablePartitionedIterator`.
+ */
+private[spark] class DefaultWritablePartitionedIterator[K, V](it: Iterator[((Int, K), V)])
```

Comment:
Is there any suggestion for a better class name?

Reply:
I think we can just turn the trait into the class if there's only one implementation now.

Reply:
ok ~

Reply:
turn the trait into the class will fail MiMa tests:
```
[error] spark-core: Failed binary compatibility check against org.apache.spark:spark-core_2.12:3.0.0! Found 1 potential problems (filtered 1184)
[error]  * declaration of interface org.apache.spark.util.collection.WritablePartitionedIterator is class org.apache.spark.util.collection.WritablePartitionedIterator in current version; changing interface to class breaks client code
[error]    filter with: ProblemFilters.exclude[IncompatibleTemplateDefProblem]("org.apache.spark.util.collection.WritablePartitionedIterator")
```

Can you give me some guidance on how to solve this problem? @Ngone51  @HyukjinKwon

Reply:
8835ecc add a `IncompatibleTemplateDefProblem` exclude filter to `MimaExcludes.scala`, GA passed now. @Ngone51 @HyukjinKwon